src/detection/packages/packages_haiku.c

```diff
@@ -0,0 +1,13 @@
+#include "packages.h"
+
+#include "common/io/io.h"
+
+void ffDetectPackagesImpl(FFPackagesResult* result, FFPackagesOptions* options)
+{
+    // TODO: Use the Package Kit C++ API instead (would account for disabled packages)
+
+    if (!(options->disabled & FF_PACKAGES_FLAG_PKG_BIT))
+        result->pkg = ffPackagesGetNumElements(FASTFETCH_TARGET_DIR_ROOT "/system/packages", false);
+    if (!(options->disabled & FF_PACKAGES_FLAG_PKGSRC_BIT))
+        result->pkgsrc = ffPackagesGetNumElements(FASTFETCH_TARGET_DIR_ROOT "/boot/home/config/packages", false);
```

Comment:
Usage of `pkg` and `pkgsrc` here doesn't really make sense. These are the same kind of packages (Haiku packages, "hpkg"), but rather system packages and user packages, respectively.

I see that there is such a distinction already for other packaging systems (e.g. for flatpak [here](https://github.com/fastfetch-cli/fastfetch/blob/f77a9c185df400eaaceedcad4294c75b5853133d/src/detection/packages/packages.h#L16-L17)). So I think it would make sense to add `haikuSystem` and `haikuUser` there as well.

Reply:
Yes, I just used that because I wasn't sure how it was categorized, but that'd be better.
Ideally we'd query the Package Kit because some packages could be disabled.